agent/bench-scripts/postprocess/user-benchmark-wrapper

```diff
@@ -0,0 +1,25 @@
+#!/usr/bin/env python
+
+import sys, json, os, logging 
+
+benchmark_run_dir=sys.argv[1]
+benchmark_duration=sys.argv[2]
+benchmark_summary={}
+
+# Creates a user-benchmark-summary.json if not created by the benchmark run and adds info about the duration if not provided by callee
+if os.path.exists('%s/user-benchmark-summary.json' %benchmark_run_dir) and os.path.getsize("%s/user-benchmark-summary.json" %benchmark_run_dir) !=0:
```

Comment:
It would be great to try to apply the DRY principle here.  Here is an example of what is possible:

```
#!/usr/bin/env python

import sys, json, os, logging

benchmark_run_dir = sys.argv[1]
benchmark_duration = int(sys.argv[2])
benchmark_summary = {}

open('%s/user-benchmark-summary.json' % benchmark_run_dir, 'r+') as data_file:
    try:
        data = json.load(data_file)
    except Exception as e:
        data = {}
    else:
        if duration in data:
            # JSON file has a duration field, so we don't have to do anything.
            return 0
    # At this point we may or may not have a dictionary of data loaded from
    # the JSON file, but we do know that if we did load some JSON it did not
    # have a duration field in it.
    data['duration'] = benchmark_duration
    data['duration_units'] = "sec"

    if data_file.tell() == 0:
        # Either there was nothing in the file, or they didn't create the
        # file.
        logging.warn("benchmark run didn't create a %s/user-benchmark-summary.json or the file is empty", benchmark_run_dir)

    # Prepare to re-write the file with an updated JSON document.
    data_file.seek(0, os.SEEK_SET)
    data_file.truncate(0)
    json.dump(benchmark_summary, data_file, sort_keys=True, indent=4, separators=(',', ':'))
```

Note that in the example where we catch the exception from loading the JSON file, we'll want to differentiate between bad JSON and an empty JSON file.  If you can modify that to catch the proper exceptions instead of the broad `Exception`, that would be great.